swift-gecko
===========

Workflow for parallel genome comparison using the Swift parallel scripting system.

Reference: Oscar Torreno and Oswaldo Trelles. Breaking the computational barriers of pairwise genome comparison. BMC Bioinformatics 2015, 16:250  doi:10.1186/s12859-015-0679-9

Requirements
============

This uses the tools:
- GECKO: Binaries available at http://bitlab-es.com/gecko/
- Swift: http://swift-lang.org/main/ (installation guide, documentation, downloads and general information about the Swift parallel scripting language)

Workflow invocation
===================

Run the following command in the directory where the input files are placed:

```
swift -config swift.conf allvsall.swift -l=100 -s=65 -wl=12 -fl=1
```

Note that the binaries/applications needed to execute the workflow are already in the /bin directory of this repository. For Swift to recognize them as executable programs, it is necessary to add this directory to the system path:

```
export PATH=<path to /bin>:$PATH
```

Importing provenance
====================

Once the workflow execution has finished, you can import the provenance generated by Swift to a SQLite database using the following command:

```
swiftlog -import-provenance <run directory>
```

If you want to import domain annotations generated by the workflow output files, you can run:

```
fasta_extractor <database name> <script run_id>
INF_extractor <database name> <script run_id>
```

and this information will be stored in the database annotation tables. Note that in the last two cases you must inform the script_run_id that is recorded in the database and that you want to relate the domain data. For this, you must first query the database to retrieve the id.




